mcx/distributions/constraints.py

```diff
@@ -175,14 +177,65 @@ def __call__(self, x):
         return np.all(x > 0, axis=-1) & (x_sum <= 1) & (x_sum > 1 - 1e-6)
 
 
+class _IndependentConstraint(Constraint):
+    def __str__(self):
+        return "an event is valid only if all its independent entries are valid."
+
+    def __init__(self, base_constraint, ndims):
+        assert isinstance(base_constraint, Constraint)
+        assert isinstance(ndims, int)
+        assert ndims >= 0
+
+        self.base_constraint = base_constraint
+        self.ndims = ndims
+        super().__init__()
+
+    def __call__(self, value):
+        result = self.base_constraint(value)
+        if self.ndims == 0:
+            return result
+        elif np.ndim(result) < self.ndims:
+            raise ValueError(
+                (f"Expected value.dim() >= {self.ndims}" " but got {np.ndims{result}}")
+            )
+        result = result.all(-1)
+        return result
+
+
+class _PositiveDefinite(Constraint):
+    event_dim = 2
+
+    def __call__(self, x):
+        jnp = numpy if isinstance(x, (numpy.ndarray, numpy.generic)) else np
+        symmetric = jnp.all(jnp.all(x == jnp.swapaxes(x, -2, -1), axis=-1), axis=-1)
+        # check for the smallest eigenvalue is positive
+        positive = jnp.linalg.eigh(x)[0][..., 0] > 0
+        return symmetric & positive
+
+
+class _LowerCholesky(Constraint):
```

Comment:
Don't forget to remove `LowerCholesky` at the end if we choose not to implement it.